### PR TITLE
adding custodian recipe

### PR DESCRIPTION
**Pull request by brandonkrull**

### CODE REVIEW

File: recipes/custodian/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+{% set sha256 = "30bb40a3eaa7dbbbfe01de74434d38ad4cdf4b5898b123b00631e9dcb194b5fa" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  md5: {{ sha256 }}
```
Comment (synapticarbors) [inline recipes/custodian/meta.yaml]: this should be `sha256` not `md5`

File: recipes/custodian/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+  run:
```
Comment (synapticarbors) [inline recipes/custodian/meta.yaml]: It looks like you are missing some requirements:

https://github.com/materialsproject/custodian/blob/master/setup.py#L19

File: recipes/custodian/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - python
+
+test:
+  imports:
+    - {{ name }
```
Comment (synapticarbors) [inline recipes/custodian/meta.yaml]: Please don't use jinja template variables in the import tests.

File: recipes/custodian/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  md5: {{ sha256 }}
+
+build:
```
Comment (synapticarbors) [inline recipes/custodian/meta.yaml]: It looks like this package defines a number of entry points that should be included here:
https://github.com/materialsproject/custodian/blob/master/setup.py#L44-L50

https://conda.io/docs/building/meta-yaml.html#python-entry-points

File: recipes/custodian/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+  run:
+    - python
+
+test:
```
Comment (synapticarbors) [inline recipes/custodian/meta.yaml]: Once you add the entry points, you will want to test each of them:

https://conda.io/docs/building/meta-yaml.html#test-commands

**Comment by brandonkrull** [comment]
thanks for the updates! i will go ahead and fix these and try again.

**Comment by brandonkrull** [comment]
i checked out the failed CI builds and they all failed when there is no import of 'six' module during custodian import. i can't reproduce that issue locally, and it looks as if 'six' is installed cleanly earlier on in the build. suggestions @synapticarbors

**Comment by synapticarbors** [comment]
monty, pyyaml and six should all be run requirements, not build requirements.

File: recipes/custodian/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  run:
+    - python
+    - monty
+    - six
+    - pyyaml
+
+test:
+  imports:
+    - custodian
+    - custodian.test_custodian
```
Comment (synapticarbors) [inline recipes/custodian/meta.yaml]: This import looks to be invalid. Please have, but it looks like it is in `custodian.tests.test_custodian`. That said, typically you don't have to import the tests module.

**Pull request closed without merging**